examples/upsert-resource-multiple/.fn-config/fn-config.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: config.kubernetes.io/v1alpha1
+kind: UpsertResourceList
```

Comment:
Why `UpsertResourceList` vs the existing `ResourceList`? Aren't the schemas identical?

Reply:
`UpsertResourceList` is identical to `v1.List` schema-wise. `ResourceList` has more fields than `v1.List`. e.g. `functionConfig` and `results`.
Another reason is that we want the function to work with both kpt and kustomize. To work with kustomize, a function must support a unique GVK as its functionConfig, since kustomize uses this GVK to identify functions.

When we build a new function we want to make it work with both kpt and kustomize. And it won't hurt the interoperability between kpt and kustomize.

I guess we can simply use `kpt fn source | sed` to get an `UpsertResourceList`.

Reply:
> UpsertResourceList is identical to v1.List schema-wise. ResourceList has more fields than v1.List. e.g. functionConfig and results.

If that's the case, why don't we just use `v1.List`?

We should not proliferate separate Kinds for each function. If they have the same semantic meaning and schema, they should be the same Kind.

> Another reason is that we want the function to work with both kpt and kustomize. To work with kustomize, a function must support a unique GVK as its functionConfig, since kustomize uses this GVK to identify functions.

Other functions accept ConfigMaps generically, that seems strange. In general, I think this is a poor design choice we should reconsider.

Reply:
We will be using v1.List.